app/assets/javascripts/sufia/relationships_table.es6

```diff
@@ -0,0 +1,190 @@
+  // error message.
+  bindKeyEvents() {
+    let $form = this.$element.parents("form");
+
+    $form.on("keyup keypress", "input.related_works_ids", function(event) {
+      let $row = $(this).parents("tr:first");
+      let key_code = event.keyCode || event.which;
+
+      // ENTER key was pressed, wait for keyup to click the Add button
+      if (key_code === 13) {
+        if (event.type == "keyup") {
+          $row.find(".btn-add-row").click();
+        }
+        event.preventDefault();
+        return false;
+      }
+
+      // ESC key was pressed, clear the input field and hide the error
+      if (key_code === 27 && event.type == "keyup") {
+        $(this).val("");
+        $row.find(".message").addClass("hidden");
+      }
+    });
+  }
+
+  // Set the warning message related to the appropriate row in the table
+  setWarningMessage($row, message) {
+    $row.find(".message.has-warning").text(message).removeClass("hidden");
+  }
+
+  // Clone an existing row, inserting it immediately before the last row,
+  // and set the clone and original row to appropriate state
+  cloneRow($row) {
+    let $table = $row.parents("table:first");
+    let $clone = $row.clone();
+    $clone.find(".btn-add-row").addClass("hidden");
+    $clone.find(".btn-remove-row").removeClass("hidden");
+    $clone.insertBefore($row);
+
+    // finally, empty the "add" row input value
+    $row.find("input.new-form-control").val("");
+    return $clone;
+  }
+
+  // Call the server, then call the appropriate callbacks to handle success and
+  // errors
+  callAjax(args) {
+    let $this = this;
+    // Send only the IDs in this table that aren't in the midst of being
+    // "removed"
+    let data = args.table.find("input.related_works_ids:not('.removing')").serialize();
+    $.ajax({
+        type: 'patch',
+        url: args.url,
+        dataType: 'json',
+        data: data
+      })
+      .done(function(json) {
+        args.on_success($this, args, json);
+      })
+      .fail(function(jqxhr, status, err) {
+        args.on_error($this, args, jqxhr, status, err);
+      });
+  }
+
+  // Set a warning message to alert the user on an error
+  handleError($this, args, jqxhr, status, err) {
+    args.row.find('input.removing').removeClass('removing');
+    let message = jqxhr.statusText;
+    if(jqxhr.responseJSON){
+      message = jqxhr.responseJSON.description;
+    }
+    $this.setWarningMessage(args.row, message);
+  }
+
+  // Remove the row when the API returns this type of success
+  handleRemoveRowSuccess($this, args, json) {
+    args.row.remove();
+  }
+
+  // Add a new row to the table, query the server for details about the work to
+  // set the title and link for the new work that was added. Hide the input
+  // field and display the title and edit button
+  handleAddRowSuccess($this, args, json) {
+    let $clone = $this.cloneRow(args.row);
+    let $title = $clone.find("a.title:first");
```

Comment:
can the DOM element be encapsulated by a new ES6 object? Can `cloneRow` be return a `RelationshipRow`?